native/common/jp_classhints.cpp

```diff
@@ -405,6 +405,39 @@ class JPConversionJavaObjectAny : public JPConversion
 	}
 } _javaObjectAnyConversion;
 
+class JPConversionJavaNumberAny : public JPConversion
+{
+public:
+
+	virtual JPMatch::Type matches(JPMatch &match, JPJavaFrame *frame, JPClass *cls, PyObject *pyobj) override
+	{
+		JP_TRACE_IN("JPConversionJavaNumberAny::matches");
+		JPValue *value = PyJPValue_getJavaSlot(pyobj);
+		if (value == NULL || frame == NULL || value->getClass() == NULL
+				|| value->getClass() == frame->getContext()->_boolean
+				|| value->getClass() == frame->getContext()->_char)
+			return match.type = JPMatch::_none;
+		match.conversion = &_javaObjectAnyConversion;
+		JPClass *oc = value->getClass();
+		if (oc == NULL)
+			return match.type = JPMatch::_none;
+		if (oc->isPrimitive())
+			return match.type = JPMatch::_implicit;
+		if (oc == cls)
+			return match.type = JPMatch::_exact;
+		bool assignable = frame->IsAssignableFrom(oc->getJavaClass(), cls->getJavaClass()) != 0;
+		return match.type = (assignable ? JPMatch::_implicit : JPMatch::_none);
+		JP_TRACE_OUT;
+	}
+
+	virtual jvalue convert(JPJavaFrame *frame, JPClass* cls, PyObject* pyobj) override
+	{
+		jvalue  v;
+		v.l = 0;
```

Comment:
I dont understand, I thought we want to convert the Java number. So why this is alsways zero?

Reply:
This one is a wierd edge case.  We do the match here but there is no convert method, but because it must be implemented because it is a pure virtual function we have to fill in a stub method.   If you read a few lines up you will see that it instead points to javaObjectAny for the actually conversion.  

I will switch this to be derived from another class so that it has better clarity.

Reply:
I correct this in numbers2 by changing the base type to JavaObjectAnyConversion so that the stub method can be dropped.